File: csharp/Axiom/RandomPlayer.cs
using System;
using System.Collections.Generic;

namespace Axiom
{
    public class RandomPlayer : Player
    {
        private Random random = new Random();

        // Constructor    
        RandomPlayer(int playerNum, int oppNum) : base(playerNum, oppNum) { }

        override
        public int ChooseMove(BoardGame board)
        {
            // Returns the next move that this player wants to make
            List<int> t = board.LegalMoves(this);
            return t[random.Next(t.Count)];
        }
    } 
}

File: java/RandomPlayer.java
import java.util.*;

/**
 * Translated from the original python written by Christine Alvorado
 */

public class RandomPlayer extends Player {

    // Constructor    
    public RandomPlayer(int playerNum, int oppNum) {
        super(playerNum, oppNum);
    }

    public int chooseMove(BoardGame board) {
        // Returns the next move that this player wants to make
        int move = -1;
        ArrayList<Integer> t = board.legalMoves(this);
        move = t.get((int)(Math.random() * t.size()));
        return move;
    }
} 
